core/src/main/java/com/sequenceiq/cloudbreak/converter/RecipeRequestToRecipeConverter.java

```diff
@@ -49,6 +77,23 @@ public Recipe convert(RecipeRequest json) {
         return recipe;
     }
 
+    private void setOrganization(RecipeRequest json, Recipe recipe) {
+        IdentityUser identityUser = authenticatedUserService.getCbUser();
+        try {
+            transactionService.required(() -> {
+                Set<Organization> usersOrganizations = organizationService.retrieveForUser(identityUser);
+                Organization organization = organizationService.get(json.getOrganizationId());
+                if (!usersOrganizations.contains(organization)) {
```

Comment:
Does the organisation has "hashCode" and "equals"

Reply:
yes it does